yorkie/src/main/kotlin/dev/yorkie/core/Client.kt

```diff
@@ -461,6 +488,38 @@ public class Client @VisibleForTesting internal constructor(
         }
     }
 
+    /**
+     * Removes the given [document].
+     */
+    public fun removeAsync(document: Document): Deferred<Boolean> {
+        return scope.async {
+            require(isActive) {
+                "client is not active"
+            }
+            val attachment = attachments.value[document.key] ?: run {
+                YorkieLogger.e("Client.remove", "document is not attached")
+                return@async false
+            }
+
+            document.setStatus(DocumentStatus.Removed)
```

Comment:
shouldn't it be updated after we recieve a successful response?

Reply:
it seems updating document's status to removed is also done in `Document.applyChangePack`